src/classes/CW_AdminControllerTest.cls

```diff
@@ -0,0 +1,199 @@
+@isTest
+private class CW_AdminControllerTest {
+
+/**
+* File Name   :    CW_AdminControllerTest.cls
+* Description :    Test Class for CW_AdminController.cls
+*
+* Modification Log
+* ============================================================================
+* Ver Date       Author         Modification
+* --- ---------- -------------- --------------------------
+* 0.1 24/04/2014 Chris Yorkston	Initial version
+*
+*
+*/
+
+    @isTest static void testAdminControllerUpdateAudienceGroupMembers() {
+    	
+    	// setup data
+		system.assertEquals(null, CustomSettingUtil.getRunBatchDate());
+		
+		CW_CampaignListSetting__c batchRunDateRecord = new CW_CampaignListSetting__c();
+		batchRunDateRecord.Name = ConstantUtil.BATCH_RUN_DATE;
+		//schedule start date cannot be today or in the past
+		Date runBatchDate = Date.today().addDays(1);
+		batchRunDateRecord.Date_Value__c = runBatchDate;
+		insert batchRunDateRecord;
+		
+		// asset that we can retrieve run batch date from custom setting
+		system.assertEquals(runBatchDate, CustomSettingUtil.getRunBatchDate());
+		
+		List<CW_Campaign__c> camps = TestDataUtil.getCampaigns(3);
+		insert camps;
+		
+		List<CW_Schedule__c> schedules = new List<CW_Schedule__c>();
+		
+		// schedule is created with evaluation start date (runBatchDate -2 days) and evaluation end date (runBatchDate +2 days)
+		schedules.add( TestDataUtil.getSchedule(camps[0].Id, runBatchDate) );
+		
+		// schedule is created with evaluation start date and evaluation end date before runBatchDate
+		schedules.add( TestDataUtil.getSchedule(camps[2].Id, runBatchDate.addMonths(2)) );
+		
+		insert schedules;
+		
+		//CW_Audience_Group__c audienceGroup = TestDataUtil.createAudienceGroup(camp.Id);
+		List<CW_Audience_Group__c> audienceGroups = TestDataUtil.getAudienceGroups(camps);
+		insert audienceGroups;
+		
+		List<Contact> contacts = TestDataUtil.getContacts(3);
+		insert contacts;
+		
+		/*
+		// 10/07/2014 CY - no longer required to insert AGS records manually as they are created by trigger
+		List<CW_Audience_Group_Schedule__c> audienceGroupSchedules = new List<CW_Audience_Group_Schedule__c>();
+		audienceGroupSchedules.add( TestDataUtil.createAudienceGroupSchedule(audienceGroups[0].Id, schedules[0].Id) );
+		audienceGroupSchedules.add( TestDataUtil.createAudienceGroupSchedule(audienceGroups[1].Id, schedules[1].Id) );
+		audienceGroupSchedules.add( TestDataUtil.createAudienceGroupSchedule(audienceGroups[2].Id, schedules[2].Id) );
+		insert audienceGroupSchedules;
+		*/
+		Set<Id> audienceGroupIds = new Set<Id>();
+		for (CW_Audience_Group__c ag :audienceGroups) audienceGroupIds.add(ag.Id);
+		List<CW_Audience_Group_Schedule__c> audienceGroupSchedules = 
+			[select Id, Name, Audience_Group__c, Schedule__c from CW_Audience_Group_Schedule__c where Audience_Group__c in :audienceGroupIds];
```

Comment:
Problem: Unbound SOQL query. Reason: a SOQL query can return up to 50,000 records. If this number is exceeded a runtime exception is thrown. Suggestion: cap the max number of records returned adding a LIMIT clause to the query.